src/rules.c

```diff
@@ -4135,6 +4141,54 @@ solver_addrecommendsrules(Solver *solv)
   solv->recommendsrules_end = solv->nrules;
 }
 
+/* add rules to exclude solvables provided by lower
+ * precedence repositories */
+void solver_addstrictrepopriorules(struct s_Solver *solv, Map *addedmap)
+{
+  Pool *pool = solv->pool;
+	Solvable *s;
+	Id p, p2, pp2;
+	Map priomap;
+	int max_prio;
+
+	map_init_clone(&priomap, addedmap);
+	solv->strictrepopriorules = solv->nrules;
+
+	FOR_POOL_SOLVABLES(p)
+	{
+		if (!MAPTST(&priomap, p))
+		  	continue;
+
+		s = pool->solvables + p;
+		max_prio = s->repo->priority;
+		FOR_PROVIDES(p2, pp2, s->name)
+		{
+		Solvable *s2 = pool->solvables + p2;
+		if (s->name != s2->name)
```

Comment:
this probably never happens in the case of conda repodata, right?

Reply:
It's good to do anyway as the strict repo prio code will probably be used by other distros as well.